pkg/kv/kvnemesis/validator.go

```diff
@@ -1073,6 +1231,12 @@ func validScanTime(
 	defer func() { _ = iter.Close() }()
 	iter.SeekGE(storage.EncodeMVCCKey(storage.MVCCKey{Key: span.Key}))
 	for ; iter.Valid(); iter.Next() {
+		// TODO(during review): is this correct? Should be but have Erik check.
+		if hasPoint, _ := iter.HasPointAndRange(); !hasPoint {
```

Comment:
We're not setting `IterKeyTypePointsAndRanges` on the iterator, so this has no effect. But if we did, then yeah, this logic would be correct. Don't we also need to handle range key deletions?

Btw, if we set appropriate bounds on the iterator we wouldn't have to do the bounds checks below.